### ISSUE TITLE
multipart HTML messages not displaying

### ISSUE THREAD

**Issue by chrisdolan** [thunderbird/thunderbird-android#1368]
Since updating to the 5.108 prerelease, I can no longer see the body of _some_ multipart HTML emails. The header appears intact, and the messages appear fine in other mail clients. Plain text mail is fine, as are many HTML emails. I think the failing ones all have inline attachments (graphics) but I'm not 100% sure of that. When I turn on debugging and "adb logcat" I see the following exception appear each time I try to view a problematic HTML email.

Devices:
 1) Google Nexus 5X, latest Marshmallow 6.0.1
 2) Sony Tablet Z2, Marshmallow 6.0.1, February update.

Perhaps unrelated, I did have to force k-9 to request the IMAP folder list after the update to see all of my folders besides just the INBOX, Trash, etc. I tried "Clear local messages" for my Inbox, but that did not change the behavior. I have not yet dared to try "Clear messages" or "Recreate data" on the account.

  05-10 13:51:34.286 D/k9      ( 9362): MessageView displaying message MessageReference{accountUuid='9f5ff35a-c63f-4071-853b-9c8aec540551', folderName='INBOX', uid='319609', flag=null}
  05-10 13:51:34.362 E/k9      ( 9362): Unable to getTextFromPart
  05-10 13:51:34.362 E/k9      ( 9362): com.fsck.k9.mail.MessagingException: File not found
  05-10 13:51:34.362 E/k9      ( 9362):     at com.fsck.k9.mailstore.FileBackedBody.getInputStream(FileBackedBody.java:32)
  05-10 13:51:34.362 E/k9      ( 9362):     at com.fsck.k9.mail.internet.MimeUtility.decodeBody(MimeUtility.java:1021)
  05-10 13:51:34.362 E/k9      ( 9362):     at com.fsck.k9.mail.internet.MessageExtractor.getTextFromPart(MessageExtractor.java:78)
  05-10 13:51:34.362 E/k9      ( 9362):     at com.fsck.k9.mailstore.LocalMessageExtractor.buildHtml(LocalMessageExtractor.java:153)
  05-10 13:51:34.362 E/k9      ( 9362):     at com.fsck.k9.mailstore.LocalMessageExtractor.extractTextAndAttachments(LocalMessageExtractor.java:117)
  05-10 13:51:34.362 E/k9      ( 9362):     at com.fsck.k9.mailstore.LocalMessageExtractor.decodeMessageForView(LocalMessageExtractor.java:442)
  05-10 13:51:34.362 E/k9      ( 9362):     at com.fsck.k9.ui.message.DecodeMessageLoader.loadInBackground(DecodeMessageLoader.java:46)
  05-10 13:51:34.362 E/k9      ( 9362):     at com.fsck.k9.ui.message.DecodeMessageLoader.loadInBackground(DecodeMessageLoader.java:15)
  05-10 13:51:34.362 E/k9      ( 9362):     at android.content.AsyncTaskLoader.onLoadInBackground(AsyncTaskLoader.java:312)
  05-10 13:51:34.362 E/k9      ( 9362):     at android.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:69)
  05-10 13:51:34.362 E/k9      ( 9362):     at android.content.AsyncTaskLoader$LoadTask.doInBackground(AsyncTaskLoader.java:66)
  05-10 13:51:34.362 E/k9      ( 9362):     at android.os.AsyncTask$2.call(AsyncTask.java:295)
  05-10 13:51:34.362 E/k9      ( 9362):     at java.util.concurrent.FutureTask.run(FutureTask.java:237)
  05-10 13:51:34.362 E/k9      ( 9362):     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
  05-10 13:51:34.362 E/k9      ( 9362):     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
  05-10 13:51:34.362 E/k9      ( 9362):     at java.lang.Thread.run(Thread.java:818)
  05-10 13:51:34.362 E/k9      ( 9362): Caused by: java.io.FileNotFoundException: /storage/emulated/0/Android/data/com.fsck.k9/files/9f5ff35a-c63f-4071-853b-9c8aec540551.db_att/3569: open failed: ENOENT (No such file or directory)
  05-10 13:51:34.362 E/k9      ( 9362):     at libcore.io.IoBridge.open(IoBridge.java:452)
  05-10 13:51:34.362 E/k9      ( 9362):     at java.io.FileInputStream.<init>(FileInputStream.java:76)
  05-10 13:51:34.362 E/k9      ( 9362):     at com.fsck.k9.mailstore.FileBackedBody.getInputStream(FileBackedBody.java:30)
  05-10 13:51:34.362 E/k9      ( 9362):     ... 15 more
  05-10 13:51:34.362 E/k9      ( 9362): Caused by: android.system.ErrnoException: open failed: ENOENT (No such file or directory)
  05-10 13:51:34.362 E/k9      ( 9362):     at libcore.io.Posix.open(Native Method)
  05-10 13:51:34.362 E/k9      ( 9362):     at libcore.io.BlockGuardOs.open(BlockGuardOs.java:186)
  05-10 13:51:34.362 E/k9      ( 9362):     at libcore.io.IoBridge.open(IoBridge.java:438)
  05-10 13:51:34.362 E/k9      ( 9362):     ... 17 more

**Comment by chrisdolan** [comment]
BTW, I sideloaded both of my devices back to 5.010 and the problem is gone. Not at all a surprise, but I thought I'd state that explicitly.

**Comment by cketti** [comment]
Can you please attach the source of the message that isn't displayed properly?

**Comment by chrisdolan** [comment]
I can no longer reproduce this problem with 5.111. Sorry for the long delay in this response...

**Comment by chrisdolan** [comment]
Aha, I was too quick. I found a message that fails to display in 5.111. I edited it slightly to remove my email address and the bounce address. [failing-message.eml.zip](https://github.com/k9mail/k-9/files/462554/failing-message.eml.zip)

**Comment by hslabbert** [comment]
I've seen similar issues: some multipart messages fail to display.

Sony Xperia Z3
Android 6.01
K9 5.111

Has been happening since I got 5.010.  I do have OpenKeychain set up (the main reason I jumped up to the alphas).

The funny thing is that preview text in the main message list view displays, but no content shows in the message view.

Interestingly enough, the two samples I could remember for this were both programmatically-generated: one from Netflix and one from Twitter.  Lightly redacted and attached.

[message-samples.zip](https://github.com/k9mail/k-9/files/476126/message-samples.zip)
